File: veraAPI/veraAPI/Models/Forms/TravelAuthForm.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using VeraAPI.Models.Tools;

namespace VeraAPI.Models.Forms
{
    public class TravelAuthForm : BaseForm
    {
        public string Preparer { get; set; } = string.Empty;            // String
        public string FirstName { get; set; } = string.Empty;           // String
        public string LastName { get; set; } = string.Empty;            // String
        public string Phone { get; set; } = string.Empty;               // Int
        public string Email { get; set; } = string.Empty;               // String
        public string EventTitle { get; set; } = string.Empty;          // String
        public string Location { get; set; } = string.Empty;            // String
        public string TravelBegin { get; set; } = string.Empty;         // DateTime
        public string TravelEnd { get; set; } = string.Empty;           // DateTime
        public string DistVehicle { get; set; } = string.Empty;         // Bool
        public string RegistrationCost { get; set; } = string.Empty;    // Decimal
        public string Airfare { get; set; } = string.Empty;             // Decimal
        public string RentalCar { get; set; } = string.Empty;           // RentalCar
        public string Fuel { get; set; } = string.Empty;                // Fuel
        public string ParkingTolls { get; set; } = string.Empty;        // ParkingTolls
        public string Mileage { get; set; } = string.Empty;             // Mileage
        public string Lodging { get; set; } = string.Empty;             // Lodging
        public string PerDiem { get; set; } = string.Empty;             // PerDiem
        public string TravelDays { get; set; } = string.Empty;            // TravelDays
        public string FullDays { get; set; } = string.Empty;            // FullDays
        public string Misc { get; set; } = string.Empty;                // Misc
        public string MiscExplain { get; set; } = string.Empty;         // MiscExplain
        public string TotalEstimate { get; set; } = string.Empty;       // TotalEstimate
        public string Advance { get; set; } = string.Empty;             // Advance
        public string AdvanceAmount { get; set; } = string.Empty;       // AdvanceAmount
        public string AdvanceDate { get; set; } = string.Empty;
        public string AdvanceStatus { get; set; } = string.Empty;
        public string Policy { get; set; } = string.Empty;              // Policy
        public string SubmitterSig { get; set; } = string.Empty;        // Integer
        public string SubmitDate { get; set; } = string.Empty;
        public string RecapRegistrationCost { get; set; } = string.Empty;   // RecapRegistrationCost
        public string RecapAirfare { get; set; } = string.Empty;        // RecapAirfare
        public string RecapRentalCar { get; set; } = string.Empty;      // RecapRentalCar
        public string RecapFuel { get; set; } = string.Empty;           // RecapFuel
        public string RecapParkingTolls { get; set; } = string.Empty;   // RecapParkingTolls
        public string RecapMileage { get; set; } = string.Empty;        // RecapMileage
        public string RecapMileageAmount { get; set; } = string.Empty;        // RecapMileageAmount
        public string RecapLodging { get; set; } = string.Empty;        // RecapLodging
        public string RecapPerDiem { get; set; } = string.Empty;        // RecapPerDiem
        public string RecapTravelDays { get; set; } = string.Empty;       // RecapTravelDays
        public string RecapFullDays { get; set; } = string.Empty;       // RecapFullDays
        public string RecapMisc { get; set; } = string.Empty;           // RecapMisc
        public string TotalRecap { get; set; } = string.Empty;          // TotalRecap
        public string RecapDate { get; set; } = string.Empty;
        public string RecapStatus { get; set; } = string.Empty;
        public string TotalReimburse { get; set; } = string.Empty;      // TotalReimburse
        public string ApprovalStatus { get; set; } = string.Empty;      // Integer - {0 Denied Red} {1 Approved Green} {2 Pending Yellow}
        public string ApprovalDate { get; set; } = string.Empty;
        public string DHApproval { get; set; } = string.Empty;          // Integer - {0 Denied Red} {1 Approved Green} {2 Pending Yellow}
        public string DHApprovalDate { get; set; } = string.Empty;
        public string DHID { get; set; } = string.Empty;                // Integer department head userID
        public string DHEmail { get; set; } = string.Empty;
        public string GMApproval { get; set; } = string.Empty;          // Integer - {0 Denied Red} {1 Approved Green} {2 Pending Yellow}
        public string GMApprovalDate { get; set; } = string.Empty;
        public string GMID { get; set; } = string.Empty;                // Integer general manager userID
        public string GMEmail { get; set; } = string.Empty;
        public string CloseDate { get; set; } = string.Empty;
        public string CloseStatus { get; set; } = string.Empty;
        public string MailMessage { get; set; } = string.Empty;

        public TravelAuthForm() { }

        public TravelAuthForm(int formDataID)
        {
            this.FormDataID = formDataID;
        }
    }
}

File: vera-ui/src/app/classes/travel-auth-form.ts


export class AuthForm {
  Preparer = false;      // Preparer
  FirstName: string;    // FirstName
  LastName: string;    // LastName
  Phone: number;      // Phone
  Email: string;    // Email
  EventTitle: string;    // EventTitle
  Location: string;    // Location
  TravelBegin: Date;        // TravelBegin
  TravelEnd: Date;        // TravelEnd
  DistVehicle: boolean;     // DistVehicle
  RegistrationCost: number;   // RegistrationCost
  Airfare: number;   // Airfare
  RentalCar: number;   // RentalCar
  Fuel: number;   // Fuel
  ParkingTolls: number;   // ParkingTolls
  Mileage: number;   // Mileage
  Lodging: number;   // Lodging
  PerDiem: number;   // PerDiem
  TravelDays:number  // days of travel
  FullDays: number;  // FullDays of travel
  Misc: number;  // Misc
  MiscExplain: string;    // MiscExplain
  TotalEstimate: number;  // TotalEstimate
  Advance: boolean;     // Advance
  AdvanceAmount: number;  // AdvanceAmount
  AdvanceDate: Date;
  AdvanceStatus: number; // approval status of the advance
  Policy: boolean;     // Policy
  SubmitterSig: number;    // SubmitterSig
  SubmitDate: Date;
  RecapRegistrationCost: number;  // RecapRegistrationCost
  RecapAirfare: number;  // RecapAirfare
  RecapRentalCar: number;  // RecapRentalCar
  RecapFuel: number;  // RecapFuel
  RecapParkingTolls: number;  // RecapParkingTolls
  RecapMileage: number;  // RecapMileage
  RecapMileageAmount: number;  // RecapMileage
  RecapLodging: number;  // RecapLodging
  RecapPerDiem: number;  // RecapPerDiem
  RecapTravelDays: number;  // RecapTravelDays
  RecapFullDays: number;  // RecapFullDays
  RecapMisc: number;  // RecapMisc
  TotalRecap: number;  // TotalRecap
  RecapDate: Date;
  RecapStatus: number; // whether the recap has been approved or not
  TotalReimburse: number;  // TotalReimburse
  ApprovalStatus: string;    // ApprovalStatus
  ApprovalDate: Date;
  DHApproval: string;     // Department head approval DHApproval
  DHApprovalDate: Date;
  DHID: number;    // Department head ID DHID
  DHEmail: string;
  GMApproval: string;     // GM approval GMApproval
  GMApprovalDate: Date;
  GMID: number;   // GMID
  GMEmail: string;
  CloseDate: Date;
  CloseStatus: number;
  MailMessage: string; // message to be sent to the traveler
}
